apps/whale-api/src/module.api/poolswap.pathfinding.service.ts

```diff
@@ -135,6 +135,10 @@ export class PoolSwapPathFindingService {
     const poolPairPaths: SwapPathPoolPair[][] = []
 
     for (const path of allSimplePaths(this.tokenGraph, fromTokenId, toTokenId)) {
+      if (path.length > 4) {
```

Comment:
Would add a quick explanation of this magic number here

```suggestion
      // Max 3 hops, e.g. A --> B --> C --> D
      if (path.length > 4) {
```

Reply:
Updated!